src/js-api.test.ts

```diff
@@ -47,9 +56,10 @@ describe(`deploy command`, () => {
   it(`requires env vars when deploying to self hosted buckets`, async () => {
     setMocks({
       staticWebSettings(settings) {
-        settings.staticFiles.microfrontendProxy.environments.prod.useBaseplateHosting =
+        const newSettings = cloneDeep(settings);
```

Comment:
The tests were influencing each other by modifying the defaultSettings object, so I switched to cloning them to ensure the tests are isolated.